src/test/osd/osdcap.cc

```diff
@@ -270,6 +270,11 @@ TEST(OSDCap, BasicRW) {
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", false, true, false, true));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, false, true, false));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, true, true, true));
+
+  // class whitelist flag works
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", false, true, false, true, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, false, true, false, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, true, true, true, true));
```

Comment:
Yeah, we want something a bit different:

1. if you have 'allow x' or 'class-read' or 'class-write', then you can 
execute any method on any *whitelisted* class.  We'll whitelist existing 
stuff by, so there is effectively no net change here (no new permissions 
and nothing is lost).

2. if you have 'allow class foo', then you can use class foo regardless of 
whether it is whitelisted or not (provided the osd is loading it). 

@athanatos right?